sei-db/sc/memiavl/snapshot.go

```diff
@@ -685,9 +1030,35 @@ func SequentialReadAndFillPageCache(filePath string) error {
 	if err != nil {
 		return err
 	}
+
+	totalSize := fileInfo.Size()
+	fmt.Printf("[PREFETCH] Starting to prefetch file: %s (size: %d MB)\n", filePath, totalSize/(1024*1024))
+
+	// Mmap the file to apply madvise hints
+	// This tells the kernel to:
+	// 1. Read sequentially (MADV_SEQUENTIAL) - enables aggressive readahead
+	// 2. Keep in cache (MADV_WILLNEED) - prioritize retention
+	// This helps prevent eviction when write buffers compete for memory
+	if totalSize > 0 {
+		data, err := unix.Mmap(int(f.Fd()), 0, int(totalSize), unix.PROT_READ, unix.MAP_SHARED)
+		if err == nil {
+			// Tell kernel this will be read sequentially - enables aggressive readahead
+			_ = unix.Madvise(data, unix.MADV_SEQUENTIAL)
+			// Tell kernel we need this data soon - start readahead immediately
+			_ = unix.Madvise(data, unix.MADV_WILLNEED)
+			// Unmap after setting hints - the hints persist on the underlying pages
+			defer unix.Munmap(data)
+			fmt.Printf("[PREFETCH] Applied madvise hints (SEQUENTIAL + WILLNEED) to %s\n", filePath)
+		} else {
+			fmt.Printf("[PREFETCH] Warning: mmap failed for %s: %v (continuing with read)\n", filePath, err)
```

Comment:
I don't quite understand why we need this step? Can we explain  it a bit more? Shouldn't we already opened and mmap the data? Why do we need to do it again here